### PR TITLE
Add overlay for enabling logs

### PR DESCRIPTION
**Pull request by adambkaplan**
# Changes

- Added a component to enable the logs feature with PVC-based storage.
- Added an overlay which deploys Results with a local database, and logs enabled with PVC storage.
- Added a README which documents how to deploy with a kustomization and contribute new kustomizations.

/kind documentation

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
The logs feature with PVC storage can be deployed and tested through a new "logs-local-db" kustomization.
```

### CODE REVIEW

**Comment by adambkaplan** [comment]
Related to #440

File: config/components/logs-file/kustomization.yaml
Comment (sayan-biswas) [inline config/components/logs-file/kustomization.yaml:1]: YAML lint complains in this file in build test.
` error    too many blank lines (1 > 0)  (empty-lines)`

Comment (adambkaplan) [reply config/components/logs-file/kustomization.yaml:1]: fixed

**Review by sayan-biswas — APPROVED** [review]

**Pull request merged**